### PR TITLE
Add Databricks sensor check partition and delta table events

### PR DESCRIPTION
**Pull request by thcidale0808**
This PR propose the creation of two new sensors within the Databricks provider package:
* Table Partition Sensor: Sense if the given partition is present in the Databricks delta table.
* Delta Table Event Sensor: Sense if there are new events for a given table. It takes a timestamp to check whether the table events are new.
Both sensors leverage the existing DatabricksSQLHook to query the information from Databricks.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by ferruzzi** [comment]
Failing static checks, please run the pre-commit tests and fix those.

File: airflow/providers/databricks/sensors/databricks.py
Comment (ferruzzi) [inline airflow/providers/databricks/sensors/databricks.py]: Nitpicking:  I'd suggest changing this to `table_name`.

File: airflow/providers/databricks/sensors/databricks.py
Comment (ferruzzi) [inline airflow/providers/databricks/sensors/databricks.py]: Nitpick:  Maybe include the expected format of the timestamp?

File: tests/providers/databricks/sensors/test_databricks.py
Comment (ferruzzi) [inline tests/providers/databricks/sensors/test_databricks.py]: Consider making the timestamp string a global so you can also reuse it in lines 64, 77, and 100

File: airflow/providers/databricks/sensors/databricks.py
Comment (thcidale0808) [reply airflow/providers/databricks/sensors/databricks.py]: Fixed

File: tests/providers/databricks/sensors/test_databricks.py
Comment (thcidale0808) [reply tests/providers/databricks/sensors/test_databricks.py]: Fixed

File: airflow/providers/databricks/sensors/databricks.py
Comment (thcidale0808) [reply airflow/providers/databricks/sensors/databricks.py]: Fixed.

**Comment by thcidale0808** [comment]
@ferruzzi your comments make sense. I've pushed the fixes.

**Comment by ferruzzi** [comment]
Nice.  You'll need a committer's approval to merge it, but looks good to me.   :+1:

File: airflow/providers/databricks/sensors/databricks.py
Comment (alexott) [inline airflow/providers/databricks/sensors/databricks.py:88]: In the Base Sensor you use catalog & schema, and here you use only database (schema). Please unify that

File: airflow/providers/databricks/sensors/databricks.py
Comment (alexott) [inline airflow/providers/databricks/sensors/databricks.py:129]: I would generalize this into the `DatabricksSqlSensor` and just pass different conditions

File: airflow/providers/databricks/sensors/databricks.py
Comment (alexott) [inline airflow/providers/databricks/sensors/databricks.py:103]: Also, It would be useful to keep last "known" version, and compare it with the current - like I did for Delta sharing: https://github.com/apache/airflow/pull/22692/files#diff-219de7033c7aa02bc63b2bb0b8f6441f2d076bc48f52af129ed70d2e132b6944R98

**Review by alexott — CHANGES_REQUESTED** [review]
You also need to update provider.yaml to register sensors, plus add documentation

**Pull request closed without merging**
